lib/src/services/app_coin_services.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import '../config/app_config.dart';
+import '../config/constants.dart';
+import '../models/coins.dart';
+import 'coin_service.dart';
+
+class AppCoinService extends CoinService {
+  AppCoinService(http.Client client) : super(client);
+
+  Future<List<Coins>> getCoinsFromAPI(
+      {required String currency,
+      required int start,
+      required int limit,
+      required bool sparkline}) async {
+    var url = Uri.parse(
+        '${AppConfig.instance.getValue(AppConstants.HOST_NAME)}' +
+            '${AppConfig.instance.getValue(AppConstants.COIN_PATH)}' +
+            '${AppConfig.instance.getValue(AppConstants.MARKET_PATH)}' +
+            '${AppConfig.instance.getValue(AppConstants.CURRENCY)}' +
+            '$currency' +
+            '${AppConfig.instance.getValue(AppConstants.ORDER)}' +
+            '${AppConfig.instance.getValue(AppConstants.LIMIT)}' +
+            '$limit' +
+            '${AppConfig.instance.getValue(AppConstants.START)}' +
+            '$start' +
+            '${AppConfig.instance.getValue(AppConstants.SPARKLINE)}' +
+            '$sparkline');
+
```

Comment:
inproper implementation of String interpolation.

Reply:
fixed